mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -0,0 +1,191 @@
+ */
+public class MlflowClient {
+  private final MlflowProtobufMapper mapper = new MlflowProtobufMapper();
+  private final MlflowHttpCaller httpCaller;
+
+  /** Returns a default client based on the MLFLOW_TRACKING_URI environment variable. */
+  public MlflowClient() {
+    this(getDefaultTrackingUri());
+  }
+
+  /** Instantiates a new client using the provided tracking uri. */
+  public MlflowClient(String trackingUri) {
+    this(getHostCredsProviderFromTrackingUri(trackingUri));
+  }
+
+  /**
+   * Creates a new MlflowClient. Users should prefer constructing ApiClients via
+   * {@link #MlflowClient()} ()} or {@link #MlflowClient(String)} if possible.
+   */
+  public MlflowClient(MlflowHostCredsProvider hostCredsProvider) {
+    httpCaller = new MlflowHttpCaller(hostCredsProvider);
+  }
+
+  public Run getRun(String runUuid) {
+    URIBuilder builder = newURIBuilder("runs/get").setParameter("run_uuid", runUuid);
+    return mapper.toGetRunResponse(httpCaller.get(builder.toString())).getRun();
+  }
+
+  public RunInfo createRun(long experimentId, String sourceName) {
+    CreateRun.Builder request = CreateRun.newBuilder();
+    request.setExperimentId(experimentId);
+    request.setSourceName(sourceName);
+    request.setSourceType(SourceType.LOCAL);
+    request.setStartTime(System.currentTimeMillis());
+    String username = System.getProperty("user.name");
+    if (username != null) {
+      request.setUserId(System.getProperty("user.name"));
+    }
+    return createRun(request.build());
+  }
+
+  public RunInfo createRun(CreateRun request) {
+    String ijson = mapper.toJson(request);
+    String ojson = doPost("runs/create", ijson);
+    return mapper.toCreateRunResponse(ojson).getRun().getInfo();
+  }
+
+  public List<RunInfo> listRunInfos(long experimentId) {
+    SearchRuns request = SearchRuns.newBuilder().addExperimentIds(experimentId).build();
+    String ijson = mapper.toJson(request);
+    String ojson = doPost("runs/search", ijson);
+    return mapper.toSearchRunsResponse(ojson).getRunsList().stream().map(Run::getInfo)
+      .collect(Collectors.toList());
+  }
+
+  public List<Experiment> listExperiments() {
+    return mapper.toListExperimentsResponse(httpCaller.get("experiments/list"))
+      .getExperimentsList();
+  }
+
+  public GetExperiment.Response getExperiment(long experimentId) {
+    URIBuilder builder = newURIBuilder("experiments/get")
+      .setParameter("experiment_id", "" + experimentId);
+    return mapper.toGetExperimentResponse(httpCaller.get(builder.toString()));
+  }
+
+  public long createExperiment(String experimentName) {
+    String ijson = mapper.makeCreateExperimentRequest(experimentName);
+    String ojson = httpCaller.post("experiments/create", ijson);
+    return mapper.toCreateExperimentResponse(ojson).getExperimentId();
+  }
+
+  public void logParam(String runUuid, String key, String value) {
+    doPost("runs/log-parameter", mapper.makeLogParam(runUuid, key, value));
+  }
```

Comment:
Should we add getParam too? It should match getMetric in returning a string if we do a float for that.